### DISCUSSION TITLE
[Question] How to unit test when using @feathersjs/mongodb?

### DISCUSSION THREAD

**Discussion by JPStrydom** [feathersjs/feathers#3443] [Q&A]
### Question

When running a feathers app with MongoDB, how can one go about unit testing without having to connect to an actual MongoDB?

The documentation around [test-database-setup](https://feathersjs.com/guides/basics/testing#test-database-setup) (and [this](https://blog.feathersjs.com/testing-feathers-applications-eaf1a323d2d9) article) mentions that one can specify a local file path for `nedb` in the `config/test.json` file like so:
``` json
{
  "nedb": "../test/data"
}
```
The documentation and article also mentions that _"The same thing can be done with connection strings for other databases."_, but this does not seem to work for MongoDB. 

I tried setting the following in the `config/test.json` file:
```json
{
  "mongodb": "../test/data"
}
```
Which MongoDB doesn't seem to like.

Is there a recommended way by Feathers to mock out MongoDB? Looks like there are options like [mongodb-memory-server](https://github.com/nodkz/mongodb-memory-server) and [mongo-mock](https://www.npmjs.com/package/mongo-mock), but don't want to sink too much time into those if Feathers has a better way of achieving this.

Thanks!

### System configuration

**Module versions** (especially the part that's not working):
- `"@feathersjs/adapter-commons": "^5.0.23"`
- `"@feathersjs/feathers": "^5.0.23"`
- `"@feathersjs/mongodb": "^5.0.23"`
- `"mongodb": "^6.4.0",`

**NodeJS version**: v20

**Comment by daffl** [comment]
We are using https://www.npmjs.com/package/mongodb-memory-server for testing, the setup is here: https://github.com/feathersjs/feathers/blob/dove/packages/mongodb/test/index.test.ts#L143
